### PR TITLE
feat(data): standalone support for ng add @ngrx/data

### LINKED ISSUE #3935: Feature: add standalone support to ng add for NgRx Data
### Which @ngrx/* package(s) are relevant/related to the feature request?

data

### Information

This feature adds support for "ng add" to NgRx Data in a standalone application. We have example usage in the schematics of other NgRx packages.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by va-stefanek**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently ng add do not have standalone for @ngrx/data package

Closes #3935 

Add standalone configuration support for ng add command for @ngrx/data package

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks @va-stefanek. The snapshot test should be committed also to fix the failing test

**Comment by va-stefanek** [comment]
@brandonroberts done :)

**Review by brandonroberts — APPROVED** [review]

File: modules/data/schematics/ng-add/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+};
+"
+`;
+
+exports[`Data ng-add Schematic Migration of ngrx-data Data ng-add Schematic for standalone application provides data without entityConfig 1`] = `
+"import { ApplicationConfig } from '@angular/core';
+import { provideEntityData, withEffects } from '@ngrx/data';
+
+export const appConfig: ApplicationConfig = {
+  providers: [provideEntityData(withEffects())]
```
Comment (markostanimirovic) [inline modules/data/schematics/ng-add/__snapshots__/index.spec.ts.snap:19]: The `provideEntityData` function expects `entityConfig` as the required argument. Check the function signature here: https://github.com/ngrx/platform/blob/main/modules/data/src/provide-entity-data.ts#L240

If the `entityConfig` option is `false`, we can pass an empty object:

```ts
  providers: [provideEntityData({}, withEffects())],
```

File: modules/data/schematics/ng-add/__snapshots__/index.spec.ts.snap [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+"
+`;
+
+exports[`Data ng-add Schematic Migration of ngrx-data Data ng-add Schematic for standalone application provides default data setup 1`] = `
+"import { ApplicationConfig } from '@angular/core';
+import { provideEntityData, withEffects } from '@ngrx/data';
+import { entityConfig } from './entity-metadata';
+
+export const appConfig: ApplicationConfig = {
+  providers: [provideEntityData(withEffects(), entityConfig)]
```
Comment (markostanimirovic) [inline modules/data/schematics/ng-add/__snapshots__/index.spec.ts.snap:30]: `entityConfig` should be the first argument:

```ts
  providers: [provideEntityData(entityConfig, withEffects())]
```

**Review by markostanimirovic — CHANGES_REQUESTED** [review]

**Review by markostanimirovic — APPROVED** [review]

**Pull request merged**